canonical-web-and-design/maas-ui
# feat(ui): add ability to select VMs in LXD VM table

## Done

- Add ability to select VMs in LXD VM table

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to the project tab of a LXD KVM and check that you can select and unselect VMs
- Check that the action dropdown and delete buttons are disabled with a tooltip if no VMs are selected
- Select some VMs then go to the machine list
- Check that no machines are selected in the machine list
- Select some of the pod's VMs from the machine list and go back to the LXD project tab
- Check that no VMs are selected in the VM list

## Fixes

Fixes #2253
